crates/adapters/src/catalog.rs

```diff
@@ -225,6 +229,9 @@ pub trait SerCursor {
     /// Serialize current key. Panics if invalid.
     fn serialize_key(&mut self, dst: &mut Vec<u8>) -> AnyResult<()>;
 
+    /// Serialize current key. Panics if invalid.
+    fn serialize_arrow_key(&mut self, dst: &mut ArrowBuilder) -> AnyResult<()>;
```

Comment:
> Is there a more abstract Trait we can use instead of ArrowBuilder?


This is bit of a desperation move we decided to do here since the given APIs &mut [u8] and Vec<u8> don't really work that well for arbitrary formats... We'll have to rework this API a bit but I'm not really familiar enough to say how it should look like.